boa3/internal/compiler/codegenerator/codegeneratorvisitor.py

```diff
@@ -1023,8 +1025,9 @@ def visit_Attribute(self, attribute: ast.Attribute) -> GeneratorData:
                 symbol = None
                 result_type = None
                 symbol_index = None
+                is_load_context_instance_variable = isinstance(attribute.ctx, ast.Load) and isinstance(attr, Variable) and value_data.symbol_id == 'self'
```

Comment:
Checking if the symbol_id is `self` worries me a bit. Don't class variables in a class method enter this flow?

i.e:
```python
class Example: 
     var_1 = 'example' 
  
     @classmethod
     def test(cls) -> str: 
         var_1 = 'example 2' 
         if len(cls.var_1) > 0: 
             return cls.var_1 
         return var_1 
```